File: examples/node/helloworld/helloworld.js
/**
 * Neutral TS Hello World Node.js example
 * https://github.com/FranBarInstance/neutralts-docs/
 */

const path = require('path');

// Import NeutralIpcTemplate (IPC client for Node.js)
const NeutralIpcTemplate = require('../neutral_ipc_template/NeutralIpcTemplate');

// The schema contains among other things the data and variables for the template
const schema = {
    data: {
        hello: "Hello World"
    }
};

// Determine the template path
const templatePath = path.join(__dirname, 'template.ntpl');

// Create an instance of NeutralIpcTemplate
const ipcTemplate = new NeutralIpcTemplate(templatePath, schema);

// Render the template
(async () => {
    try {
        const contents = await ipcTemplate.render();

        // Print the rendered content, in other cases contents will be sent to output according to framework.
        console.log(contents);

    } catch (error) {
        console.error('Error rendering template:', error.message);
        process.exit(1);
    }
})();

File: examples/go/helloworld/helloworld.go
/*
Neutral TS Hello World IPC example
https://github.com/FranBarInstance/neutralts-docs/
*/

package main

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"

	"neutral_ipc_template"
)

func main() {
	// The schema contains among other things the data and variables for the template
	schemaDict := map[string]interface{}{
		"data": map[string]string{
			"hello": "Hello World",
		},
	}

	// Determine the template path
	dir, err := os.Getwd()
	if err != nil {
		fmt.Println("Error getting current directory:", err)
		os.Exit(1)
	}
	templatePath := filepath.Join(dir, "template.ntpl")

	// Marshal schema to JSON string
	schemaJSON, err := json.Marshal(schemaDict)
	if err != nil {
		fmt.Println("Error marshaling schema:", err)
		os.Exit(1)
	}

	// Create an instance of NeutralIpcTemplate
	ipcTemplate := neutral_ipc_template.NewNeutralIpcTemplate(templatePath, string(schemaJSON))

	// Render the template
	contents := ipcTemplate.Render()

	// Print the rendered content, in other cases contents will be sent to output according to framework.
	fmt.Println(contents)
}
